js/apps/account-ui/src/applications/Applications.tsx

```diff
@@ -154,8 +155,11 @@ const Applications = () => {
             </DataListItemRow>
 
             <DataListContent
+              id={`content-${application.clientId}`}
               className="pf-u-pl-4xl"
-              aria-label={t("applicationDetails")}
+              aria-label={t("applicationDetails", {
+                clientId: application.clientId,
```

Comment:
This is currently not accepted by the label for "applicationDetails", so you'll have to update it as well, or perhaps define a new label that has `{{clientId}}` inside of it. Either way it has to be human readable and descriptive.